### ISSUE TITLE
@available causes build to fail with Xcode < 9

### ISSUE THREAD

**Issue by jmroot** [libsdl-org/SDL#6601]
The `@available` syntax was added in [Xcode 9.0](https://developer.apple.com/library/archive/releasenotes/DeveloperTools/RN-Xcode/Chapters/Introduction.html#//apple_ref/doc/uid/TP40001051-CH1-SW878). 5892ed4 added unconditional use of it, so trying to build with an older Xcode version results in an error like this:

```
…/SDL2-2.26.0/src/joystick/iphoneos/SDL_mfijoystick.m:642:9: error: unexpected '@' in program
    if (@available(macOS 10.16, *)) {
        ^
```

You can probably just wrap the check in `#if __has_builtin(__builtin_available)`, since anything that doesn't have that builtin is not macOS 11 or later.

**Comment by jmroot** [comment]
This seems to work:
```diff
--- src/joystick/iphoneos/SDL_mfijoystick.m.orig	2022-11-18 04:16:08.000000000 +1100
+++ src/joystick/iphoneos/SDL_mfijoystick.m	2022-11-24 07:04:48.000000000 +1100
@@ -639,11 +639,16 @@ static int
 IOS_JoystickInit(void)
 {
 #if defined(__MACOSX__)
+#if _SDL_HAS_BUILTIN(__builtin_available)
     if (@available(macOS 10.16, *)) {
         /* Continue with initialization on macOS 11+ */
     } else {
         return 0;
     }
+#else
+    /* No @available, must be an older macOS version */
+    return 0;
+#endif
 #endif
 
     @autoreleasepool {
```
Do you want me to open a PR, or is this sufficient?

**Comment by slouken** [comment]
Yes, please create a PR. There may be a bunch of places where this is necessary.
